aws/resource_aws_ram_resource_share.go

```diff
@@ -0,0 +1,241 @@
+
+	if aws.StringValue(resourceShare.Status) != ram.ResourceShareStatusActive {
+		log.Printf("[WARN] RAM resource share (%s) delet(ing|ed), removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", resourceShare.Name)
+	d.Set("allow_external_principals", resourceShare.AllowExternalPrincipals)
+
+	if err := d.Set("tags", tagsToMapRAM(resourceShare.Tags)); err != nil {
+		return fmt.Errorf("Error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsRamResourceShareUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ramconn
+
+	d.Partial(true)
+
+	if d.HasChange("name") || d.HasChange("allow_external_principals") {
+		request := &ram.UpdateResourceShareInput{
+			ResourceShareArn:        aws.String(d.Id()),
+			Name:                    aws.String(d.Get("name").(string)),
+			AllowExternalPrincipals: aws.Bool(d.Get("allow_external_principals").(bool)),
+		}
+
+		log.Println("[DEBUG] Update RAM resource share request:", request)
+		_, err := conn.UpdateResourceShare(request)
+		if err != nil {
+			if isAWSErr(err, ram.ErrCodeUnknownResourceException, "") {
+				log.Printf("[WARN] No RAM resource share by ARN (%s) found", d.Id())
+				d.SetId("")
+				return nil
+			}
+			return fmt.Errorf("Error updating RAM resource share %s: %s", d.Id(), err)
+		}
+
+		d.SetPartial("name")
+		d.SetPartial("allow_external_principals")
+	}
+
+	if d.HasChange("tags") {
+		// Reset all tags to empty set
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		c, r := diffTagsRAM(tagsFromMapRAM(o), tagsFromMapRAM(n))
+
+		if len(r) > 0 {
+			_, err := conn.UntagResource(&ram.UntagResourceInput{
+				ResourceShareArn: aws.String(d.Id()),
+				TagKeys:          tagKeysRam(r),
+			})
+			if err != nil {
+				return fmt.Errorf("Error deleting RAM resource share tags: %s", err)
+			}
+		}
+
+		if len(c) > 0 {
+			input := &ram.TagResourceInput{
+				ResourceShareArn: aws.String(d.Id()),
+				Tags:             c,
+			}
+			_, err := conn.TagResource(input)
+			if err != nil {
+				return fmt.Errorf("Error updating RAM resource share tags: %s", err)
+			}
+		}
+
+		d.SetPartial("tags")
+	}
+
+	d.Partial(false)
+
+	return resourceAwsRamResourceShareRead(d, meta)
+}
+
+func resourceAwsRamResourceShareDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ramconn
+
+	deleteResourceShareInput := &ram.DeleteResourceShareInput{
+		ResourceShareArn: aws.String(d.Id()),
+	}
+
+	log.Println("[DEBUG] Delete RAM resource share request:", deleteResourceShareInput)
+	_, err := conn.DeleteResourceShare(deleteResourceShareInput)
+	if err != nil {
+		if isAWSErr(err, ram.ErrCodeUnknownResourceException, "") {
+			return nil
+		}
+		return fmt.Errorf("Error deleting RAM resource share %s: %s", d.Id(), err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{ram.ResourceShareStatusDeleting},
```

Comment:
We may also want to include `ram.ResourceShareStatusActive` in the `Pending` list in case there is any delay the start of the process. 👍